compiler/modules/types/list.bal

```diff
@@ -257,6 +272,55 @@ function listInhabited(Context cx, SemType[] members, SemType rest, Conjunction?
     }
 }
 
+// fill out to length of newLen with filler
+function fixedLengthArrayFill(FixedLengthArray arr, int newLen, SemType filler) {
+    SemType[] initial = arr.initial;
+    if arr.fixedLength == 0 || initial[initial.length() - 1] != filler {
+        initial.push(filler);
+    }
+    arr.fixedLength = newLen;
+}
+
+function fixedArrayGet(FixedLengthArray members, int index) returns SemType {
+    int memberLen = members.initial.length();
+    int i = int:min(index, memberLen - 1);
+    return members.initial[i];
+}
+
+function listMemberAt(FixedLengthArray fixedArray, SemType rest, int index) returns SemType {
+    if index < fixedArray.fixedLength {
+        return fixedArrayGet(fixedArray, index);
+    }
+    return rest;
+} 
+
+function fixedArraySet(FixedLengthArray members, int setIndex, SemType m) {
+    int initCount = members.initial.length();
+    boolean lastMemberRepeats = members.fixedLength > initCount;
+
+    // No need to expand
+    if setIndex < initCount - (lastMemberRepeats ? 1 : 0) {
+        members.initial[setIndex] = m;
+        return;
+    }
+        
+    SemType lastMember = members.initial[initCount - 1]; 
+    foreach int i in initCount ..< setIndex + 1 {
+        members.initial.push(lastMember);
+    }
+    members.initial[setIndex] = m;
+}
+
+function fixedArrayWith(FixedLengthArray array, int index, SemType newMember) returns FixedLengthArray {
```

Comment:
I guess you are not using `replace` because it sounds like it is mutating. I am not wild about `with`. If we cannot find a good name, then maybe better to inline it, since it is for simple and only called once.

Reply:
Inlined in `list.bal` and `listProj.bal`